JuliaInterface/gap/JuliaInterface.gi

```diff
@@ -57,11 +57,74 @@ InstallMethod( CallFuncList,
     end );
 
 
-##
-##  We support '<juliaobj>[<i>]' and '<juliaobj>[<i>, <j>]' in general.
-##  This is useful for example if <juliaobj> is a tuple.
-##
-BindJuliaFunc( "getindex", "Base" );
+BindGlobal( "_JULIA_MODULE_TYPE", JuliaEvalString( "Module" ) );
+BindGlobal( "_JULIA_FUNCTION_TYPE", JuliaEvalString( "Function" ) );
+BindGlobal( "_JULIA_ISA", JuliaFunction( "isa" ) );
+
+BindGlobal( "_NewJuliaModule",
+  function( name, julia_pointer )
+    local module;
+
+    module := rec( storage := rec( ),
+                   julia_pointer := julia_pointer );
+
+    Objectify( TheTypeOfJuliaModules, module );
+
+    SetName( module, name );
```

Comment:
The name is what is printed, right? Instead of printing just, say `Main` or `JSON`, I think it'd be nicer if we printed e.g. `Julia module "Main"` or `JuliaModule("Main")` or whatever. Perhaps with a custom `ViewObj` or `ViewString` method?